discovery-provider/src/tasks/entity_manager/user.py

```diff
@@ -123,9 +124,31 @@ def create_user(params: ManageEntityParameters):
         is_current=False,
     )
 
-    if params.metadata == "v2":
+    user_metadata = None
+    try:
+        # for single tx signup
+        user_metadata = json.loads(params.metadata)["data"]
+        validate_user_metadata(
+            params.session,
+            user_record,
+            user_metadata,
+        )
+
+        user_record = update_user_metadata(
+            params.session,
+            params.redis,
+            user_record,
+            user_metadata,
+            params.web3,
+            params.challenge_bus,
+        )
+    except Exception:
+        # fallback to multi tx signup
+        pass
+
+    if params.metadata_cid == "v2":
         user_record.is_storage_v2 = True
-    else:
+    elif not user_metadata:  # update replica set case
```

Comment:
couple questions:
1. whats the old multi tx flow? a CREATE USER event followed by an UPDATE USER event?
2. if so, with the single tx flow i assume we want the cid metadata that would normally be in the UPDATE USER event to be persisted in `cid_data` so the client can query it later? 
    - in which case we either need to add special cases for different metadata formats in CREATE USER events to `expect_cid_metadata_json` and `parse_metadata` in entity manager `utils.py`
    - or
    - handle that here and persist to `cid_data` appropriately
    - feels convoluted trying to handle all possible CREATE USER cases upstream in entity_manager.py as we migrate from multi->single tx and also from storage v1->v2 so would prefer the latter, with the custom logic here
4. what are all the possible metadata values that we want to handle here? (considering old multi tx path, new single tx path, and v2)

Reply:
1. Yea, it's a create user, update for the handle, update for profile picture.
2. Yeah sg, updating it to treat this case as a user update and parse.
3. This will just be for storage v2 so I think the only cases we will have are v2 single tx and v2 multi tx.